docs/rules/array-bracket-spacing.md

```diff
@@ -18,26 +18,35 @@ var [x,y] = z;
 ## Rule Details
 
 This rule aims to maintain consistency around the spacing inside of array brackets, either by disallowing
-spaces inside of brackets between the brackets and other tokens or enforcing spaces. Brackets that are
-separated from the adjacent value by a new line are excepted from this rule, as this is a common pattern.
-  Object literals that are used as the first or last element in an array are also ignored.
+spaces inside of brackets between the brackets and other tokens or enforcing spaces.
 
 ## Options
 
-There are two options for this rule:
+This rule has a string option:
 
-* `"always"` enforces a space inside of array brackets
-* `"never"` enforces no space inside of array brackets (default)
+* `"never"` (default) disallows spaces inside array brackets
+* `"always"` enforces spaces inside array brackets
 
-Depending on your coding conventions, you can choose either option by specifying it in your configuration:
+This rule has built-in exceptions:
 
-```json
-"array-bracket-spacing": ["error", "always"]
-```
+* `"never"` allows newlines inside brackets of an array literal, as this is a common pattern
+* `"always"` does not require spaces in an empty array literal `[]`
```

Comment:
This is one aspect I tried to keep consistent across rule descriptions: Should we use singular or plural here? I.e. "[...] in an empty array literal" or "[...] in empty array literals" ? In the rule descriptions in the README I went with plural where it made sense (I don't have a fundamental reason why I chose plural, I just picked one to keep consistent). Do we want to stick with that for option descriptions as well, or should I change the rule descriptions?